README.md

```diff
@@ -33,23 +33,158 @@ Main features
 - :microscope: Tiny and tree-shakeable
 - :muscle: Supports TypeScript
 
-## Docs
+## Table of Contents
+
+- [Installation](#installation)
+- [API](#api)
+  - [connectObservable](#connectObservable)
+  - [connectFactoryObservable](#connectFactoryObservable)
+  - [distinctShareReplay](#distinctShareReplay)
+  - [SUSPENSE](#SUSPENSE)
+  - [suspend](#suspend)
+  - [suspended](#suspended)
+  - [switchMapSuspended](#switchMapSuspended)
+  - [createInput](#createInput)
+- [Examples](#examples)
+
+
+## Installation
+    npm install re-rxjs
+
+## API
+
+### connectObservable
+```ts
+const [useCounter, sharedCounter$] = connectObservable(
+  clicks$.pipe(
+    scan(prev => prev + 1, 0),
+    startWith(0),
+  )
+)
+```
+Returns a hook that provides the latest update of the observable, and an enhanced 
+version of the original observable (the one that the hook will use), which is shared 
+and "replayable".
+
+### connectFactoryObservable
+```tsx
+const [useStory, getStory$] = connectFactoryObservable(
+  (storyId: number) => getStoryWithUpdates$(storyId)
+)
+
+const Story: React.FC<{id: number}> = ({id}) => {
+  const story = useStory(id);
+
+  return (
+    <article>
+      <h1>{story.title}</h1>
+      <p>{story.description</p>
+    </article>
+  )
+}
+```
+Returns a hook that receives as arguments the same arguments as the factory function 
```

Comment:
Returns a hook that presents the latest value of an observable generated from a factory function.

Accepts: 
A factory function that returns an Observable

Returns [1 , 2]:
1. A function with the same parameters as the factory function, calling this function in a React component will create and subscribe to an observable generated by the accepted factory function. The function will then return the latest update from the observable.
2. A shared replayable version of the observable generated by the factory function that is used by the hook internally.

Reply:
need to name 1 and 2!